core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -1962,6 +1962,26 @@ private[spark] object Utils extends Logging {
     path
   }
 
+  /**
+   * Updates Spark config with properties from a set of Properties.
+   * Provided properties have the highest priority.
+   */
+  def updateSparkConfigFromProperties(conf: SparkConf, properties: Properties): Unit = {
```

Comment:
Yes with the exception of this line: sys.props.getOrElseUpdate(k, v) for which again as I said I am not sure.